opal/corelib/runtime.js

```diff
@@ -1219,10 +1215,20 @@
     //         - super (window.Object)
     //           - null
     //
-    $defineProperty(native_klass, '$$bridge', klass);
-    $set_proto(native_klass.prototype, (klass.$$super || Opal.Object).$$prototype);
-    $defineProperty(klass, '$$prototype', native_klass.prototype);
+    if (!native_klass.hasOwnProperty('$$bridge')) {
+      var ourPrototype = (klass.$$super || Opal.Object).$$prototype;
+      try {
+        $set_proto(native_klass.prototype, ourPrototype);
+      } catch(error) {
+        if (error.name != "TypeError") {
+          throw error;
+        }
```

Comment:
You think we can get to a more solid check here? e.g. by using `Opal.is_a(Opal.TypeError)` or something like that?

Reply:
I don't think there's any difference, anyway updated.